### ISSUE TITLE
loss=0 for vit_imagenet_k400 model training

### ISSUE THREAD

**Issue by charlieliu219** [taoyang1122/adapt-image-models#49]
Hi,

I've come across an issue when trying to reproduce the training results using config `vit_imagenet_k400.py` with [k400 data from mmaction](https://github.com/open-mmlab/mmaction2/blob/main/tools/data/kinetics/README.md). From the beginning to the end of the training, the loss remains 0. 

I've checked the output of the dataloader, which looks normal:

2025-08-05 13:04:54,444 - mmaction - INFO -     Images shape: torch.Size([8, 1, 3, 8, 224, 224])
2025-08-05 13:04:54,446 - mmaction - INFO -     Images dtype: torch.float32
2025-08-05 13:04:54,448 - mmaction - INFO -     Images device: cpu
2025-08-05 13:04:54,452 - mmaction - INFO -     Images range: [-2.117904, 2.640000]
2025-08-05 13:04:54,459 - mmaction - INFO -     Non-zero elements: 9633792/9633792 (100.00%)
2025-08-05 13:04:54,459 - mmaction - INFO -     Labels: tensor([[ 78],

Therefore, I put debugging statements in the `forward` method of the model `ViT_ImageNet`:

def forward(self, x):
        logger = get_root_logger()
        logger.info(f"Input tensor with {torch.count_nonzero(x)} non-zero elements")
        
        B, C, T, H, W = x.shape
        x = rearrange(x, 'b c t h w -> (b t) c h w')
        x = self.patch_embed(x)  # shape = [BT, HW, D]
        x = torch.cat([self.cls_token.to(x.dtype) + torch.zeros(x.shape[0], 1, x.shape[-1], dtype=x.dtype, device=x.device), x], dim=1) 
        x = x + self.pos_embed.to(x.dtype)

        n = x.shape[1]
        x = rearrange(x, '(b t) n d -> (b n) t d', t=self.num_frames)
        x = x + self.temporal_embedding
        x = rearrange(x, '(b n) t d -> (b t) n d', n=n)
        
        for blk in self.blocks:
            x = blk(x)  # [BT, HW+1, D]

        x = self.ln_post(x)
        x = x[:, 0]
        x = rearrange(x, '(b t) c -> b c t',b=B,t=T)
        
        x = x.unsqueeze(-1).unsqueeze(-1)  # BDTHW for I3D head

        logger.info(f"Output tensor with {torch.count_nonzero(x)} non-zero elements")

        return x

And the ouputs are like below for every single data point. 

2025-08-05 13:25:48,294 - mmaction - INFO - Input tensor with 8827 non-zero elements
2025-08-05 13:25:48,403 - mmaction - INFO - Output tensor with 0 non-zero elements

Has this ever occurred before when using only the original code and config?
